src/setfit/data.py

```diff
@@ -128,3 +128,64 @@ def create_fewshot_splits_multilabel(dataset: Dataset, sample_sizes: List[int])
             split_df = create_samples_multilabel(df, sample_size, seed)
             splits_ds[f"train-{sample_size}-{idx}"] = Dataset.from_pandas(split_df, preserve_index=False)
     return splits_ds
+
+
+def add_templated_examples(
+    dataset: Dataset,
+    candidate_labels: List[str],
+    template: str = "This sentence is {}",
+    sample_size: int = 2,
+    text_column: str = "text",
+    label_column: str = "label",
+) -> Dataset:
+    """Adds templated examples to a Dataset.
+
+    The Dataset is assumed to have a text column with the name `text_column` and a
+    label column with the name `label_column`, which contains one-hot or multi-hot
+    encoded label sequences.
+
+    Args:
+        dataset (Dataset): The Dataset to add templated examples to.
+        candidate_labels (List[str]): This list of candidate labels to be fed into
+            the template to construct examples. This should align with the
+            `label_column_name` column of `dataset`.
+        template (str, optional): The template used to turn each label into an
+            NLI-style hypothesis. This template must include a {} for the candidate
```

Comment:
This isn't quite the same as NLI modeling, so let's tweak this a bit:
```suggestion
            synthetic training example. This template must include a {} for the candidate
```